[NGT] Fix some Muon validation plots, allow for summary plots creation

Parsifal-2045 opened the pull request:
#### PR description:

This PR deals with some general fixes and improvements for Muon validation targeting both HLT and Offline for Phase 1 and Phase 2:
- Fix DQM plots filling for quantities vs dR (now active when exactly two signal `TP`s are present), for dimuon studies. This also required to extend the interface of the `TrackingParticleSelector` class with a few more getters, including `isSignalOnly()`
- Fix plots for quantities separated in barrel, overlap, and endcap regions (e.g. efficiency_vs_phi_barrel) which should now be filled with the correct information in each region. 
- Implement summary plots (efficiency vs muon collection and fake rate vs muon collection) for both HLT and Offline validation
- Slightly rework validation code to make summary plots creation possible: now there is a single `MuonMultiTrackValidator` for HLT and a few for Offline (one per group: standard, displaced, refit, ...) 
- The new `MuonMultiTrackValidator` takes as input vectors of `labels`, `associators`, and `muonHistoParameters`. These are supposed to be of the same size ~(runtime check, with an exception thrown if not)~ and in the same order ~(no check at the moment, could be an item for further development)~. [589e1f9](https://github.com/cms-sw/cmssw/pull/47692/commits/589e1f9cea9943b32b62c03d5f638b6c9cb8f7ce) implements a configuration check for all `MuonTrackValidator`s' consistency. See https://github.com/cms-sw/cmssw/pull/47692#issuecomment-2755597459 
- Remove some outdated and not used validation sequences in `HLTriggerOffline` dealing with muon collections that are no longer produced under tracking
- Simplify muon association / validation sequences for HLT and Offline

[Full talk](https://indico.cern.ch/event/1530334/contributions/6438878/subcontributions/540396/attachments/3038713/5366920/2025-03-24%20-%20Muon%20POG%20validation%20update.pdf) discussing these changes presented at the HLT Upgrade on 25/03/2025.
No physics changes are expected. DQM outputs should also remain largely the same, with the notable exclusions of plots vs dR, plots for separate regions, and the removal of outdated and always-empty plots produced by the previously mentioned sequence under `HLTriggerOffline`.

#### PR validation:

I've run the usual battery of tests locally and produced a DQM output comparison for two different ZMM workflows: one for Run 3 and one for Phase 2. The DQM comparison output is available [here](https://cernbox.cern.ch/s/6RukcqF3s6OG3rF) and was obtained using the `compareHistograms.py` utility of CMSSW.

Parsifal-2045 committed:
Fix Muon Validation dR plots filling
- Now applies only in cases where exactly two signal TPs are produced
- dR is calculated using information from those two TPs

Parsifal-2045 committed:
Add methods to access private members of TrackingParticleSelector

Parsifal-2045 committed:
Implement summary plot creation for Muons
- Muon track validator now takes a vector of input tags for label, associatormap, and muonHistoParameters (same size, same order)
- Summary plots can be used to compare efficiency and fake rate of muon objects both at the HLT and Offline
- A single MuonTrackValidator object is used at the HLT to produce summary plots comparing all collections
- Offline still split into multiple MuonTrackValidator objects, grouped per type (standard, refit, displaced, TeV, cosmic)
- Update MuonTrackValidator logging information

Parsifal-2045 committed:
Fix barrel, overlap, and endcap histograms filling criteria

Parsifal-2045 committed:
Use phase2_muon proc modifier to modify histogram parameters in muon validation

Parsifal-2045 committed:
Clean up muon validation sequences for HLT

Parsifal-2045 committed:
Clean up associators and sequences for Offline muon validation

Parsifal-2045 committed:
Remove deprecated Muon Tracking validation modules

Parsifal-2045 commented:
type ngt

cmsbuild commented:
cms-bot internal usage<!-- bot cache: {"commits":{"2b8d9b19124d972cdc2ef34dce6c277a8954316c":{"files":["HLTriggerOffline/Common/python/HLTValidationHarvest_cff.py","HLTriggerOffline/Common/python/HLTValidation_cff.py","HLTriggerOffline/Muon/python/HLTmultiTrackValidatorMuonTracks_cff.py","HLTriggerOffline/Muon/python/HLTpostProcessorMuonTrack_cfi.py"],"squashed":false,"time":1742898417},"36eab51f1ef6a9b2c95f464850be139f15c1bdb2":{"files":["Validation/RecoMuon/plugins/MuonTrackValidator.cc","Validation/RecoMuon/plugins/MuonTrackValidator.h","Validation/RecoMuon/plugins/MuonTrackValidatorBase.h","Validation/RecoMuon/python/MuonTrackValidator_cfi.py","Validation/RecoMuon/python/PostProcessorHLT_cff.py","Validation/RecoMuon/python/PostProcessor_cff.py","Validation/RecoMuon/python/histoParameters_cff.py","Validation/RecoMuon/python/muonValidationHLT_cff.py","Validation/RecoMuon/python/muonValidation_cff.py"],"squashed":false,"time":1742897566},"4ed4bfde9ffd04910e868f4055eb3fdcd65b9c29":{"files":["Validation/RecoMuon/plugins/MuonTrackValidator.cc","Validation/RecoMuon/plugins/MuonTrackValidatorBase.h","Validation/RecoMuon/python/histoParameters_cff.py"],"squashed":false,"time":1742377171},"572a06512c72046bc7578fa2a9e70fc5aeebe95f":{"files":["Validation/RecoMuon/python/muonValidationHLT_cff.py"],"squashed":false,"time":1742898399},"82830fd304af8df4dcadd4bd2f897fb732da7a5d":{"files":["Validation/RecoMuon/python/histoParameters_cff.py","Validation/RecoMuon/python/muonValidation_cff.py"],"squashed":false,"time":1742377171},"993b65cf85c6f3fdff0301afe65f0d753d7a4256":{"files":["Validation/RecoMuon/python/MuonTrackValidator_cfi.py","Validation/RecoMuon/python/associators_cff.py","Validation/RecoMuon/python/muonValidation_cff.py"],"squashed":false,"time":1742898399},"a56c581e595e5577d269866b326b07dd3e3da6d1":{"files":["SimTracker/Common/interface/TrackingParticleSelector.h"],"squashed":false,"time":1742377170},"b145ff0a1033209cdcc192855180332062023cd6":{"files":["Validation/RecoMuon/plugins/MuonTrackValidator.cc"],"squashed":false,"time":1742377171}},"emoji":{"2752094705":"+1"},"last_seen_sha":"2b8d9b19124d972cdc2ef34dce6c277a8954316c","signatures":{}} -->

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-47692/44240

cmsbuild commented:
A new Pull Request was created by  @Parsifal-2045 for master.

It involves the following packages:

- HLTriggerOffline/Common (**dqm**)
- HLTriggerOffline/Muon (**dqm**)
- SimTracker/Common (**simulation**)
- Validation/RecoMuon (**dqm**)


@antoniovagnerini, @civanch, @cmsbuild, @kpedro88, @mdhildreth, @rseidita can you please review it and eventually sign? Thanks.
@CeliaFernandez, @Fedespring, @GiacomoSguazzoni, @HuguesBrun, @VinInn, @VourMa, @abbiendi, @andrea21z, @calderona, @cericeci, @dgulhan, @echabert, @gbenelli, @jhgoh, @missirol, @mmusich, @mtosi, @prolay, @robervalwalsh, @rociovilar, @rovere, @threus, @trocino this is something you requested to watch as well.
@antoniovilela, @mandrenguyen, @rappoccio, @sextonkennedy you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

mmusich commented:
@cmsbuild, please test

cmsbuild commented:
+1

**Size**: This PR adds an extra 64KB to repository
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-b675b5/45201/summary.html
**COMMIT**: 2b8d9b19124d972cdc2ef34dce6c277a8954316c
**CMSSW**: CMSSW_15_1_X_2025-03-25-2300/el8_amd64_gcc12
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/47692/45201/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Comparison Summary



**Summary**:
 - You potentially [removed 14 lines](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_15_1_X_2025-03-25-2300+b675b5/67666/validateJR/logRootQA-events.log) from the logs
 - [Reco comparison results: 8 differences found in the comparisons](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_15_1_X_2025-03-25-2300+b675b5/67666/validateJR.html)
 - DQMHistoTests: Total files compared: 50
 - DQMHistoTests: Total histograms compared: 3907849
 - DQMHistoTests: [Total failures: 2724](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_15_1_X_2025-03-25-2300+b675b5/67666/validateJR.html)
 - DQMHistoTests: Total nulls: 288
 - DQMHistoTests: Total successes: 3904817
 - DQMHistoTests: Total skipped: 20
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: -5812.875999999997 KiB( 49 files compared)
 - DQMHistoSizes: changed ( 10224.0,... ): 0.381 KiB        Muons/RecoMuonV
 - DQMHistoSizes: changed ( 16834.0,... ): -2912.436 KiB    HLT/Muon
 - DQMHistoSizes: changed ( 16834.0,... ): -0.004 KiB       MessageLogger/Errors
 - DQMHistoSizes: changed ( 16834.0,... ): -0.004 KiB       MessageLogger/Warnings
 - DQMHistoSizes: changed ( 24834.911,... ): 0.448 KiB        HLT/Muon
 - DQMHistoSizes: changed ( 24834.911,... ): 0.232 KiB        Muons/RecoMuonV
 - DQMHistoSizes: changed ( 24834.911,... ): -0.012 KiB       MessageLogger/Errors
 - DQMHistoSizes: changed ( 24834.911,... ): -0.012 KiB       MessageLogger/Warnings
 - Checked 215 log files, 184 edm output root files, 50 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_15_1_X_2025-03-25-2300+b675b5/67666/triggerResults): found differences in 8 / 48 workflows

Parsifal-2045 committed:
Implement consistency check for Validators
- associatormap, label and muonHistoParameters must have the same size
- associatormap and label must share the same input collection (associatormap and label vectors must have the same ordering)

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-47692/44265

cmsbuild commented:
Pull request #47692 was updated. @antoniovagnerini, @civanch, @cmsbuild, @kpedro88, @mdhildreth, @rseidita can you please check and sign again.

Parsifal-2045 commented:
[589e1f9](https://github.com/cms-sw/cmssw/pull/47692/commits/589e1f9cea9943b32b62c03d5f638b6c9cb8f7ce) implements a configuration check for all `MuonTrackValidator`s' consistency. In the python fragments for HLT and Offline validation `associatormap`, `labels`, and `muonHistoParameters` are required to have the same size in all validators. Furthermore, the associators listed in `associatormap` must follow the same order as the tracks in `labels`, for this reason there is a further check to make sure that each associator uses the same track input as the label it corresponds to

cmsbuild commented:
+1

**Size**: This PR adds an extra 28KB to repository
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-b675b5/45231/summary.html
**COMMIT**: 589e1f9cea9943b32b62c03d5f638b6c9cb8f7ce
**CMSSW**: CMSSW_15_1_X_2025-03-26-2300/el8_amd64_gcc12
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/47692/45231/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Comparison Summary



**Summary**:
 - You potentially [removed 11 lines](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_15_1_X_2025-03-26-2300+b675b5/67694/validateJR/logRootQA-events.log) from the logs
 - [Reco comparison results: 0 differences found in the comparisons](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_15_1_X_2025-03-26-2300+b675b5/67694/validateJR.html)
 - DQMHistoTests: Total files compared: 50
 - DQMHistoTests: Total histograms compared: 3907849
 - DQMHistoTests: [Total failures: 2780](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_15_1_X_2025-03-26-2300+b675b5/67694/validateJR.html)
 - DQMHistoTests: Total nulls: 288
 - DQMHistoTests: Total successes: 3904761
 - DQMHistoTests: Total skipped: 20
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: -5812.875999999997 KiB( 49 files compared)
 - DQMHistoSizes: changed ( 10224.0,... ): 0.381 KiB        Muons/RecoMuonV
 - DQMHistoSizes: changed ( 16834.0,... ): -2912.436 KiB    HLT/Muon
 - DQMHistoSizes: changed ( 16834.0,... ): -0.004 KiB       MessageLogger/Errors
 - DQMHistoSizes: changed ( 16834.0,... ): -0.004 KiB       MessageLogger/Warnings
 - DQMHistoSizes: changed ( 24834.911,... ): 0.448 KiB        HLT/Muon
 - DQMHistoSizes: changed ( 24834.911,... ): 0.232 KiB        Muons/RecoMuonV
 - DQMHistoSizes: changed ( 24834.911,... ): -0.012 KiB       MessageLogger/Errors
 - DQMHistoSizes: changed ( 24834.911,... ): -0.012 KiB       MessageLogger/Warnings
 - Checked 215 log files, 184 edm output root files, 50 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_15_1_X_2025-03-26-2300+b675b5/67694/triggerResults): found differences in 8 / 48 workflows

civanch commented:
+1

antoniovagnerini commented:
+dqm
- spurious differences seen from unstable WF `29634.911` (open issue #45505 )
- spurious differences from unstable WF `29834.999`(open issue #47071)

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs (tests are also fine). This pull request will now be reviewed by the release team before it's merged. @antoniovilela, @rappoccio, @mandrenguyen, @sextonkennedy (and backports should be raised in the release meeting by the corresponding L2)
